src/lib/db.ts

```diff
@@ -488,7 +488,12 @@ export class DB {
         problem: getProblemPath(this.stores),
         classes: user.classHashesForProblemPath(getProblemPath(this.stores)),
         originDoc: documentModel.key,
-        content
+        content,
+        // LTI fields
+        platform_id: user.portal,
+        context_id: user.classHash,
+        resource_link_id: user.offeringId,
+        resource_url: user.offeringUrl || ""
```

Comment:
I'm not entirely sure what offeringUrl is here.
In LTI terms the resource_url should probably be the URL to the problem.
Perhaps this is what offeringUrl is computing but I'd guess not.

In the portal this would be the url of the resource (external activity in the data modeling below the UI).
In the collaborative-learning environment you could probably just use window.location (minus any parameters)